### PR TITLE
Farming hud improvements

### PR DESCRIPTION
**Pull request by kevinthegreat1**
Add coins per hour.  
Fix the mouse not unlocking when the option is turned off.  
Refactor item price calculations.  

<details><summary>Farming Hud</summary>
<p>

Screenshot 2024-05-05 at 00 24 19

</p>
</details>

### CODE REVIEW

**Review by AzureAaron — DISMISSED** [review]

File: src/main/java/de/hysky/skyblocker/utils/ItemUtils.java [hunk 1]
```diff
@@ -207,12 +247,12 @@ public static String getHeadTexture(ItemStack stack) {
         if (!stack.isOf(Items.PLAYER_HEAD) || !stack.contains(DataComponentTypes.PROFILE)) return "";
 
         ProfileComponent profile = stack.get(DataComponentTypes.PROFILE);
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/utils/ItemUtils.java:249]:
```suggestion
        if (!stack.isOf(Items.PLAYER_HEAD) || !stack.contains(DataComponentTypes.PROFILE)) return "";

```

File: src/main/java/de/hysky/skyblocker/utils/ItemUtils.java [hunk 2]
```diff
@@ -207,12 +247,12 @@ public static String getHeadTexture(ItemStack stack) {
...
-        String texture = profile.properties().get("textures").stream()
+        if (profile == null) return "";
+
+        return profile.properties().get("textures").stream()
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/utils/ItemUtils.java:252]:
```suggestion
        return stack.get(DataComponentTypes.PROFILE).properties().get("textures").stream()
```

**Comment by kevinthegreat1** [comment]
Conflicts with #683. Will wait until after #683 is merged.

**Comment by kevinthegreat1** [comment]
Add cultivating counter and use normal counter as fallback.
Screenshot 2024-05-17 at 17 14 40

**Comment by kevinthegreat1** [comment]
Fix the merge conflict on #683’s side, so all good to review now.

File: src/main/java/de/hysky/skyblocker/skyblock/garden/FarmingHudWidget.java [hunk 3]
```diff
@@ -18,31 +19,31 @@
...
+            Map.entry("THEORETICAL_HOE_WARTS_3", "NETHER_STALK"),
+            Map.entry("FUNGI_CUTTER", "RED_MUSHROOM"),
+            Map.entry("CACTUS_KNIFE", "CACTUS"),
+            Map.entry("MELON_DICER", "MELON"),
+            Map.entry("MELON_DICER_2", "MELON"),
+            Map.entry("MELON_DICER_3", "MELON"),
+            Map.entry("PUMPKIN_DICER", "PUMPKIN"),
+            Map.entry("PUMPKIN_DICER_2", "PUMPKIN"),
+            Map.entry("PUMPKIN_DICER_3", "PUMPKIN"),
+            Map.entry("COCO_CHOPPER", "INK_SACK.3")
```
[the remark this answers is not in the document]
Comment (kevinthegreat1) [reply src/main/java/de/hysky/skyblocker/skyblock/garden/FarmingHudWidget.java]: Good catch

File: src/main/java/de/hysky/skyblocker/skyblock/garden/FarmingHud.java [hunk 4]
```diff
@@ -92,10 +88,29 @@ public static void init() {
                 .executes(Scheduler.queueOpenScreenCommand(() -> new FarmingHudConfigScreen(null)))))));
     }
 
+    private static boolean tryParseCounter(ItemStack stack, Pattern counterPattern) {
+        Matcher matcher = ItemUtils.getLoreLineIfMatch(stack, counterPattern);
+        if (matcher == null) return false;
```
[the remark this answers is not in the document]
Comment (kevinthegreat1) [reply src/main/java/de/hysky/skyblocker/skyblock/garden/FarmingHud.java:89]: Done in `118c4b7`.

**Comment by kevinthegreat1** [comment]
All fixed up, including the crash.

**Comment by kevinthegreat1** [comment]
I tried switching over to `ClientTickEvents.END_CLIENT_TICK` but since it updates slower, block per second shows 21. I tried reproducing what you're experiencing but couldn't.

**Review by AzureAaron — CHANGES_REQUESTED** [review]
The farming hud icon using the main hand stack is a bit weird.

Although the HUD is broken in the main menus and stuff (when the player has no main hand stack)

**Comment by kevinthegreat1** [comment]
Tested btw.

<details><summary>Farming Hud</summary>
<p>
Screenshot 2024-05-31 at 21 17 48
</p>
</details>

**Comment by Fluboxer** [comment]
> Just found out that the cultivating counter disappears after 100m. That got me thinking, would it be possible to calculate crops/min using bps (or blocks per min if necessary) and multiplying that by average drops per crop * (1+(fortune/100)) and then save the amount ever farmed somewhere?

is there a need to? It is still here, just not displayed

**Comment by kevinthegreat1** [comment]
All done.

**Pull request merged**